frontend/src/components/common/Resource.tsx

```diff
@@ -474,10 +484,20 @@ export function ConditionsTable(props: ConditionsTableProps) {
     return cols.filter(col => !col.hide);
   }
 
+  const [sortObj, setSortObj] = React.useState({'label': getColumns()[0].label, 'isIncreasingOrder': true});
+
+  function getSortFunc() {
+    const columnInitiatingSort = getColumns().filter((col) => {
+      return col.label === sortObj.label;
+    })[0].sortFunction;
+    return columnInitiatingSort && columnInitiatingSort(sortObj.isIncreasingOrder);
+  }
+
   return (
     <SimpleTable
-      data={(resource && resource.status && resource.status.conditions) || {}}
+      data={(resource && resource.status && resource.status.conditions.sort(getSortFunc())) || {}}
```

Comment:
You don't need to sort it here in the condution, because sorting shouldn't change the data, just the order.